framework/src/governance/lib/datazone/datazone-msk-custom-asset.ts

```diff
@@ -0,0 +1,145 @@
+import { CustomResource, Duration, RemovalPolicy, Stack } from 'aws-cdk-lib';
+import { CfnProjectMembership } from 'aws-cdk-lib/aws-datazone';
+import {
+  Effect,
+  IRole,
+  ManagedPolicy,
+  PolicyDocument,
+  PolicyStatement,
+  Role,
+  ServicePrincipal,
+} from 'aws-cdk-lib/aws-iam';
+import { IFunction } from 'aws-cdk-lib/aws-lambda';
+import { ILogGroup } from 'aws-cdk-lib/aws-logs';
+import { Construct } from 'constructs';
+import { DataZoneMSKCustomAssetProps } from './datazone-msk-custom-asset-props';
+import { Context, TrackedConstruct, TrackedConstructProps } from '../../../utils';
+import { DsfProvider } from '../../../utils/lib/dsf-provider';
+
+export class DataZoneMSKCustomAsset extends TrackedConstruct {
+  readonly createLogGroup: ILogGroup;
+  readonly createFunction: IFunction;
+  readonly createRole: IRole;
+  readonly statusLogGroup: ILogGroup;
+  readonly statusFunction: IFunction;
+  readonly statusRole: IRole;
+  readonly serviceToken: string;
+  readonly domainId: string;
+  readonly projectId: string;
+  readonly name: string;
+  readonly removalPolicy?: RemovalPolicy;
+
+  constructor(scope: Construct, id: string, props: DataZoneMSKCustomAssetProps) {
+    const trackedConstructProps: TrackedConstructProps = {
+      trackingTag: DataZoneMSKCustomAsset.name,
+    };
+
+    super(scope, id, trackedConstructProps);
+
+    this.removalPolicy = Context.revertRemovalPolicy(scope, props.removalPolicy);
+
+    // Validation logic
+    if (props.latestVersion && props.schemaVersion) {
+      throw new Error('If latestVersion is true, schemaVersionNumber cannot be provided.');
+    }
+
+    if (props.schemaArn && (props.registryName || props.schemaName)) {
+      throw new Error('If schemaArn is provided, neither registryName nor schemaName can be provided.');
+    }
+
+    if (props.registryName && !props.schemaName) {
+      throw new Error('If registryName is provided, schemaName must also be provided.');
+    }
+
+    const handlerRole = new Role(this, 'HandlerRole', {
+      assumedBy: new ServicePrincipal('lambda.amazonaws.com'),
+      managedPolicies: [
+        ManagedPolicy.fromAwsManagedPolicyName('service-role/AWSLambdaBasicExecutionRole'),
+      ],
+      inlinePolicies: {
+        DataZonePermission: new PolicyDocument({
+          statements: [
+            new PolicyStatement({
+              effect: Effect.ALLOW,
+              actions: [
+                'datazone:CreateAsset',
+                'datazone:CreateAssetType',
+                'datazone:CreateFormType',
+                'datazone:GetAssetType',
+                'datazone:GetFormType',
+                'glue:GetSchemaVersion',
+                'glue:ListSchemaVersions',
+                'datazone:CreateAssetRevision',
```

Comment:
Can you separate the Read action and Write action? Can you verify that Write action can be scoped down by requiring a tag attached to them?